runtime/lua/vim/lsp/util.lua

```diff
@@ -117,10 +119,15 @@ end
 ---@param encoding string utf-8|utf-16|utf-32|nil defaults to utf-16
 ---@return number byte (utf-8) index of `encoding` index `index` in `line`
 function M._str_byteindex_enc(line, index, encoding)
-  if encoding ~= 'utf-8' then
-    return vim.str_byteindex(line, index, not encoding or encoding ~= 'utf-32')
+  if not encoding then encoding = 'utf-16' end
+  if encoding == 'utf-8' then
+    if index then return index else return #line end
+  elseif encoding ~= 'utf-16' then
```

Comment:
This should be checking it is utf-16 no?

Reply:
Oops, yeah